homeassistant/components/fully_kiosk/button.py

```diff
@@ -0,0 +1,93 @@
+"""Fully Kiosk Browser button."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FullyKioskDataUpdateCoordinator
+from .entity import FullyKioskEntity
+
+
+@dataclass
+class FullyButtonEntityDescriptionMixin:
+    """Mixin to describe a Fully Kiosk Browser button entity."""
+
+    press_action: Callable
+
+
+@dataclass
+class FullyButtonEntityDescription(
+    ButtonEntityDescription, FullyButtonEntityDescriptionMixin
+):
+    """Fully Kiosk Browser button description."""
+
+
+BUTTONS: tuple[FullyButtonEntityDescription, ...] = (
+    FullyButtonEntityDescription(
+        key="restartApp",
+        name="Restart Browser",
+        press_action=lambda fully: fully.restartApp(),
+    ),
+    FullyButtonEntityDescription(
+        key="rebootDevice",
+        name="Reboot Device",
+        press_action=lambda fully: fully.rebootDevice(),
+    ),
+    FullyButtonEntityDescription(
+        key="toForeground",
+        name="Bring to Foreground",
+        press_action=lambda fully: fully.toForeground(),
+    ),
```

Comment:
```suggestion
    FullyButtonEntityDescription(
        key="toForeground",
        name="Bring to foreground",
        press_action=lambda fully: fully.toForeground(),
    ),
```

I've not added an entity category, as this feel like a primary entity?
As in, to active the thing? What do you think?

Reply:
I'd probably consider toForeground, toBackground, and loadStartUrl as "primary" entities

Reply:
Is that something one would expect to show up on their auto-generated Lovelace dashboards and control from their dashboard often? That is a bit of the general rule of thumb...

I suspect a lot of this integration to be mainly about device configuration/monitoring than actively controlling it from a dashboard.